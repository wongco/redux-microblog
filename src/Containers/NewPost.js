import React, { Component } from 'react';
import { connect } from 'react-redux';

import PostForm from '../Components/PostForm';
import { addPostToAPI } from '../Actions/posts';
import PropTypes from 'prop-types';

class NewPost extends Component {
  addPost = postDetails => {
    this.props.addPostToAPI(postDetails);
    this.props.history.push('/');
  };

  // returns user to home page upon canceling
  cancel = () => this.props.history.push('/');

  render() {
    return (
      <div>
        <h3>New Post</h3>
        <PostForm savePost={this.addPost} cancel={this.cancel} />
      </div>
    );
  }
}

NewPost.propTypes = {
  addPostToAPI: PropTypes.func,
  history: PropTypes.object,
  location: PropTypes.object,
  match: PropTypes.object
};

const connectedComponent = connect(
  null,
  { addPostToAPI }
);

export default connectedComponent(NewPost);
