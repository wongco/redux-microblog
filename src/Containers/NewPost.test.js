import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import NewPost from './NewPost';

it('renders without crashing', () => {
  shallow(<NewPost />);
});

it('matches snapshot for default criteria', () => {
  const wrapper = shallow(<NewPost />);
  const serialized = toJson(wrapper);
  expect(serialized).toMatchSnapshot();
});
