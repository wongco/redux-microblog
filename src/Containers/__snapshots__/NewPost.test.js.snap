// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot for default criteria 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
